### PR TITLE
feat: Person merge restrictions info to UI

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

This is confusing and not clear to users. Showing the is_identified state in the UI would help with support requests a bit.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*
Screenshot 2023-06-13 at 17 21 46
Screenshot 2023-06-13 at 17 21 30

### CODE REVIEW

**Review by xvello — APPROVED** [review]

**Review by Twixes — CHANGES_REQUESTED** [review]

File: frontend/src/scenes/persons/Person.tsx
Comment (Twixes) [inline frontend/src/scenes/persons/Person.tsx:91]: Does "applied" mean that this person can't be merged with another person at all?

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:97]: it's complicated 😅   that's part of the problem ... for example in python it's `alias(distinct_id, alias_id)` but in js its `alias(alias_id, distinct_id)`

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: It can be merged with another "none" person by using this person as the distinct_id and "none" person as anon_distinct_id or alias_id. 
It can be merged with another "applied" person only via `$merge_dangerously` events

File: frontend/src/scenes/persons/Person.tsx
Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:96]: the correct form is "cannot" . "Can not" is incorrect.

Also, its not super clear what we're referring to here. Is it the person? Or their IDs? Which Ids?

File: frontend/src/scenes/persons/Person.tsx
Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:91]: I think there could be a better phrasing for this. We're trying to say that this person cannot be merge into another person right?

A few alternatives that come to mind
`merge possible: false / true`
`merge possible` and `merge restricted`

Could also add ascii checkmarks e.g.
`✓ Merge possible` and `✗ Merge restricted`

**Review by Lior539 — CHANGES_REQUESTED** [review]
I'm super excited about this change! Thank you for doing it.

The text needs work, though

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: I don't like `possible` as it's always possible - just you'd need to use `$merge_dangerously`

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:96]: it's the person that you can't merge - but in the commands you'd use any of the distinctIDs to point to the person

File: frontend/src/scenes/persons/Person.tsx
Comment (Twixes) [reply frontend/src/scenes/persons/Person.tsx]: Okay – does this mean that another person cannot be merged into this one, or that this one cannot be merged into others?

File: frontend/src/scenes/persons/Person.tsx
Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:91]: Hmm a few more idea then:

1. `✓ Safe merge possible` : `✗ Dangerous merge only`

2. `✓ Safe merge possible` `✗ Safe merge not possible`

3. `Safe merge possible: ✓` `Safe merge possible: ✗`

I think I like the second option the most

File: frontend/src/scenes/persons/Person.tsx
Comment (Lior539) [inline frontend/src/scenes/persons/Person.tsx:97]: I thought I commented this already but couldnt find my previous comment:

Is it worth including `$anon_distinct_id` here? This is for using as an anonymous ID right? This is not super common right? In fact, I wasn't even aware you could do it.

When would our users want to do it? If I recall correctly, we dont mention this in our docs anywhere?

Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:97]: https://github.com/search?q=repo%3APostHog%2Fposthog.com%20anon_distinct_id&type=code but sure I can keep only `alias_id` can you look into the experiments doc https://posthog.com/tutorials/experiments and update that too. The how ingestion works should mention it and be unchanged probably.

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx]: this one cannot be merged into others

Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx]: we always merge into the distinct_id person we merge the alias_id/anon_id person into distinct_id person (well that's a lie there's some PoE magic possibly but that's not user visible, from their perspective it's always into distinct_id person) 
See https://posthog.com/docs/how-posthog-works/ingestion-pipeline#merging-two-persons

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: I'd want to keep the existing style of the page - so there's a keyword next to "IDs", "First seen" and then the value for it. I also wouldn't want to call it safe merge - that's a new term we don't use anywhere else. Also you can do a safe merge into this person, but you can't merge this person into anyone else.

File: frontend/src/scenes/persons/Person.tsx
Comment (Twixes) [reply frontend/src/scenes/persons/Person.tsx]: Okay, that's good to know. We need to explain this in the UI, as users generally do not look at our code reviews for details on how PostHog works. 😄 I also suggest linking to a specific section about merging in docs, as currently the link is useless. I've scrolled through all of it and there's no clear explanation what merge restrictions exist and why.

Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx]: cc @Lior539  ^ this is why I was saying we need to be super clear about that in the doc

File: frontend/src/scenes/persons/Person.tsx
Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:91]: So taking a step back here, here is my understanding of what we're trying to achieve (please correct me if Im wrong)

* We want to communicate in the app if a user has been identified or not, so that our users will better understand if they can safe merge or not

With that in mind, should we do the following:

For anonymous users, instead of:
label: `Merge restrictions: none`
tooltip `Can be used as `$anon_distinct_id` or `alias_id`, click for more info.`

Change this to:
label: `Anonymous user` (or unidentified user)
tooltip `This user has not been identified. Click for more details` and it links to https://posthog.com/docs/data/identify

And then for identified users, change it to:
label: `Identified user`
tooltip `This user has been identified. If this is a duplicate of another user, click for more details on how to merge duplicate users` and it links to https://posthog.com/docs/data/identify#how-to-handle-duplicates-of-the-same-user

What do you think?

Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: > We want to communicate in the app if a user has been identified or not, so that our users will better understand if they can safe merge or not

couple of goals:
1. For folks who are not merging users it shouldn't be confusing - "Anonymous user" to me feels confusing there, e.g. if you use backend only and haven't merged with anyone.
2. For folks who are trying to understand the ingestion warning or why their merge failed - they'd understand what was the problem
3. For folks who want to merge this user with someone they will know how to do it

Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:91]: > , e.g. if you use backend only and haven't merged with anyone

Users on the backend are identified users, no? Is it possible to have anonymous users on the backend? How? (I thought you need to include a `distinct_id` with every call on the backend => so that makes the users identified)

> Seems best to not use the word "identified" because that's a logical relation to "$identify", but it's not only that fn usage.

Makes sense

> For folks who want to merge this user with someone they will know how to do it

Good to know

Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: > Users on the backend are identified users, no? Is it possible to have anonymous users on the backend? How? (I thought you need to include a distinct_id with every call on the backend => so that makes the users identified)

No. In our code we mark a user `is_identified = true` iff any of their distinct_id's has been used as `distinct_id` for `$identify`, `$create_alias`, `$merge_dangerously`. (note this would be the case after merging https://github.com/PostHog/posthog/pull/15912 currently it's more confusing to know the exact condition 😅 ).
This means that if you are a backend only user and never send any of these events all of your users will have `is_identified = false`. This is because the frontends don't tell us that it's an anonymous ID (there's been talks about sending that info, but historically we don't). Furthermore we want to safeguard agains  aliasing badly everyone together in the backends too.

Comment (Lior539) [reply frontend/src/scenes/persons/Person.tsx:91]: ah okay! Thank you for clearing that up. I've [updated](https://github.com/PostHog/posthog.com/pull/6139) the docs to clarify this.

With this in mind, let's do the following (very similar to what you suggested):
Have `Merge restrictions: none / applied` as you suggested
For the the tooltip: `Can/Cannot be used as alias_id – click for more info.` and have that click point to the alias part of the docs -> https://posthog.com/docs/data/identify#alias-assigning-multiple-distinct-ids-to-the-same-user 

I would still exclude the "Can be used as $anonymous_id", since its not something our users would do, and it would add confusion to include it 

WDYT?

**Review by Lior539 — APPROVED** [review]

File: frontend/src/scenes/persons/Person.tsx
Comment (tiina303) [reply frontend/src/scenes/persons/Person.tsx:91]: Sounds good, will update

**Review by Twixes — APPROVED** [review]
Makes sense considering the docs clarification

**Pull request merged**
